classes/galley/Collector.inc.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * @file classes/galley/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class galley
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of galleys
+ */
+
+namespace PKP\galley;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+use PKP\plugins\HookRegistry;
+
+class Collector implements CollectorInterface
+{
+    public DAO $dao;
+
+    public ?array $publicationIds = null;
+
+    public ?array $contextIds = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    public function filterByPublicationIds(array $publicationIds): self
```

Comment:
`filterBy...` function should be able to receive `null` as a parameter, in order to reset the constraint.